character-sfx.js

```diff
@@ -342,6 +359,74 @@ class CharacterSfx {
   
     };
     _handleNarutoRun();
+
+    // combo
+    const dispatchComboSoundEvent = (combo, indexOfCombo) =>{
+      this.dispatchEvent(new MessageEvent('comboSoundEmit', {
+        data: {
+          combo: combo,
+          indexOfCombo: indexOfCombo
+        },
+      }));
+    }
+    const _handleCombo = () => {
+      let currentCombo;
+      if(this.player.hasAction('use') && this.player.getAction('use').behavior === 'sword'){
+        currentCombo = this.player.getAction('use').animation ? 
+          this.player.getAction('use').animation // sword
+          : 
+          this.player.getAction('use').animationCombo[this.player.avatar.useAnimationIndex] // silsword
+      }
+      if (currentCombo) {
+        if (currentCombo !== this.lastCombo) {
+          this.playComboTime = timestamp;
+          this.currentComboIndex = -1;
+        }
+        if (currentCombo === 'combo') { // sword
+          switch (this.currentComboIndex) {
+            case -1: { 
+              this.currentComboIndex = 0;
+            }
+            case 0: { 
+              if(timestamp - this.playComboTime >= swordAnimationOffset[this.currentComboIndex]){

```

Comment:
Given that this code is repeated, this is not a good way to write this method. There should not be any numerical state machine.

Reply:
got it.
https://github.com/webaverse/app/pull/3660/commits/1d406d2ee22549ace38c39ad3189bf2797d545dc